include/ignition/gazebo/EntityComponentManager.hh

```diff
@@ -206,14 +209,16 @@ namespace ignition
       /// \return The component associated with the key, or nullptr if the
       /// component could not be found.
       public: template<typename ComponentTypeT>
-              const ComponentTypeT *Component(const ComponentKey &_key) const;
+              const ComponentTypeT *IGN_DEPRECATED(6) Component(
+              const ComponentKey &_key) const;
```

Comment:
Windows is not happy about something here:

```
C:\Jenkins\workspace\ign_gazebo-pr-win\ws\ign-gazebo\include\ignition/gazebo/EntityComponentManager.hh(212,37): error C2059: syntax error: '' (compiling source file C:\Jenkins\workspace\ign_gazebo-pr-win\ws\ign-gazebo\src\EntityComponentManager.cc) [C:\Jenkins\workspace\ign_gazebo-pr-win\ws\build\ignition-gazebo6\src\ignition-gazebo6.vcxproj]
```

Maybe there's some weird character hiding in there? :woman_shrugging:

Reply:
It seems like it has something to do with using `IGN_DEPRECATED` in combination with methods that return a pointer. I am guessing that the `*` in the method signature is not being understood correctly when parsing occurs on windows. I removed all of the `IGN_DEPRECATED` macros for methods that return pointers in https://github.com/ignitionrobotics/ign-gazebo/commit/576efc7c5cdae78df4542b07b4249f265d4a6916 (but left the `IGN_DEPRECATED` macros for methods that do not return pointers), and these build errors no longer appear. However, I'm now seeing errors with the function definitions for the templated `View` class (https://build.osrfoundation.org/job/ign_gazebo-pr-win/2858/consoleFull#-11549300628ff58640-3599-4406-a210-216932f1748c):

```
C:\Jenkins\workspace\ign_gazebo-pr-win\ws\ign-gazebo\include\ignition/gazebo/detail/View.hh(150,27): error C2491: 'ignition::gazebo::v6::detail::View<ComponentTypeTs...>::View': definition of dllimport function not allowed (compiling source file C:\Jenkins\workspace\ign_gazebo-pr-win\ws\build\ignition-gazebo6\src\gui\ignition-gazebo6-gui_autogen\mocs_compilation.cpp) [C:\Jenkins\workspace\ign_gazebo-pr-win\ws\build\ignition-gazebo6\src\gui\ignition-gazebo6-gui.vcxproj]
```

@j-rivero do you have any thoughts about what's going on with the `IGN_DEPRECATED` macro, and also how to resolve the `definition of dllimport function not allowed` error above?

Reply:
Okay, so it sounds like the `IGN_DEPRECATED` issue has been resolved in https://github.com/ignitionrobotics/ign-gazebo/pull/856#issuecomment-896228298. However, I still believe that the `definition of dllimport function not allowed` error remains. Does anyone have any thoughts about how to handle this other build error?